src/app/events/[slug]/components/ScheduleTabs.tsx

```diff
@@ -0,0 +1,99 @@
+'use client'
+
+import { useEffect, useRef } from 'react'
+
+import { Tab, TabGroup, TabList, TabPanel, TabPanels } from '@headlessui/react'
+import theme from 'tailwindcss/defaultTheme'
+
+import type { Event } from '@/types/eventType'
+
+import { BasicCard } from '@/components/BasicCard'
+import { Heading } from '@/components/Heading'
+import { TextLink } from '@/components/TextLink'
+
+import { formatDate, formatTime } from '../utils/dateUtils'
+
+type ScheduleTabsProps = {
+  schedule: Event['schedule']
+}
+
+const { screens } = theme
+
+export function ScheduleTabs({ schedule }: ScheduleTabsProps) {
+  const tabGroupRef = useRef<HTMLDivElement>(null)
+  const hasMounted = useRef(false)
+
+  const validDays = schedule!.days.filter((day) => day.events.length > 0)
+
+  useEffect(() => {
+    hasMounted.current = true
+  }, [])
+
+  function scrollToTabGroup() {
+    if (hasMounted.current && isScreenBelowLg() && tabGroupRef.current) {
+      tabGroupRef.current.scrollIntoView({
+        behavior: 'smooth',
+        block: 'start',
+      })
```

Comment:
We already have a function `scrollToSection` in the maturity model `utils` folder, in case it can be extracted in the global `utils` reused here. Not sure, I haven't tried.

Super cool you added this by the way, smooth UX!

Reply:
Decided not to extract the `scrollToSection` function for now because it is tightly coupled to the concept of a `SectionHash`.

By the way, this made me notice that the function `scrollToSection` uses `document.querySelector`, which in React is considered an anti-pattern because it directly manipulates the DOM outside of React’s control. Instead, React encourages the use of `refs` to interact with DOM elements.

Something like

```
import { useRef } from 'react'

export function scrollToSection(sectionRef: React.RefObject<HTMLElement>) {
  if (!sectionRef.current) {
    console.error(`The referenced element does not exist`)
    return
  }

  sectionRef.current.scrollIntoView({
    block: 'start',
    behavior: 'smooth',
  })
}
```

Reply:
Indeed!